src/wallet/asyncrpcoperation_sendmany.cpp

```diff
@@ -0,0 +1,972 @@
+                
+                // Funds are removed from the value pool and enter the private pool
+                info.vpub_old += value;
+            }
+            obj = perform_joinsplit(info);
+        }
+        sign_send_raw_transaction(obj);
+        return true;
+    }
+    /**
+     * END SCENARIO #2
+     */   
+ 
+    
+    
+    /**
+     * SCENARIO #3
+     * 
+     * zaddr -> taddrs
+     *       -> zaddrs
+     * 
+     * Processing order:
+     * Part 1: taddrs and miners fee
+     * Part 2: zaddrs 
+     */
+    
+    /**
+     * SCENARIO #3
+     * Part 1: Add to the transparent value pool.
+     */
+    Object obj;
+    CAmount jsChange = 0;   // this is updated after each joinsplit
+    bool minersFeeProcessed = false;
+
+    if (t_outputs_total > 0) {
+        add_taddr_outputs_to_tx();
+        CAmount taddrTargetAmount = t_outputs_total + minersFee;
+        minersFeeProcessed  = true;
+        while (zInputsDeque.size() > 0 && taddrTargetAmount > 0) {
+            AsyncJoinSplitInfo info;
+            info.vpub_old = 0;
+            info.vpub_new = 0;
+            std::vector<JSOutPoint> outPoints;
+            int n = 0;
+            while (n++ < 2 && taddrTargetAmount > 0) {
+                SendManyInputJSOP o = zInputsDeque.front();
+                JSOutPoint outPoint = std::get<0>(o);
+                Note note = std::get<1>(o);
+                CAmount noteFunds = std::get<2>(o);
+                zInputsDeque.pop_front();
+
+                info.notes.push_back(note);
+                outPoints.push_back(outPoint);
+
+
+                LogPrint("asyncrpc", "%s: spending note (txid=%s, vjoinsplit=%d, ciphertext=%d, amount=%s)\n",
+                        getId().substr(0, 10),
+                        outPoint.hash.ToString().substr(0, 10),
+                        outPoint.js,
+                        int(outPoint.n), // uint8_t
+                        FormatMoney(noteFunds, false)
+                        );
+
+                
+                // Put value back into the value pool
+                if (noteFunds >= taddrTargetAmount) {
+                    jsChange = noteFunds - taddrTargetAmount;
+                    info.vpub_new += taddrTargetAmount;
+                } else {
+                    info.vpub_new += noteFunds;
+                }
+
+                taddrTargetAmount -= noteFunds;
+                if (taddrTargetAmount <= 0) {
+                    break;
+                }
+            }
+
+            if (jsChange > 0) {
+                info.vjsout.push_back(JSOutput());
+                info.vjsout.push_back(JSOutput(frompaymentaddress_, jsChange));
+                
+                LogPrint("asyncrpc", "%s: generating note for change (amount=%s)\n",
+                        getId().substr(0, 10),
+                        FormatMoney(jsChange, false)
+                        );
+            }
+
+            obj = perform_joinsplit(info, outPoints);
```

Comment:
Yes, tx_ is continually updated with each new joinsplit.
obj contains hex string of the raw transaction after each joinsplit and the raw notes.  It's leftover from earlier development of the code as it was useful to have that data for debugging etc.  obj is then passed to sign_send_raw_transaction().
We can remove obj and have sign_send_raw_transaction() work off tx_. Too late to make this change now, but can do it for beta 2.